yt_dlp/extractor/ciscolive.py

```diff
@@ -2,144 +2,179 @@
 
 from .common import InfoExtractor
 from ..utils import (
+    ExtractorError,
     clean_html,
-    float_or_none,
     int_or_none,
     parse_qs,
-    try_get,
+    str_or_none,
+    try_call,
     urlencode_postdata,
 )
+from ..utils.traversal import traverse_obj
 
 
 class CiscoLiveBaseIE(InfoExtractor):
-    # These appear to be constant across all Cisco Live presentations
-    # and are not tied to any user session or event
-    RAINFOCUS_API_URL = 'https://events.rainfocus.com/api/%s'
-    RAINFOCUS_API_PROFILE_ID = 'Na3vqYdAlJFSxhYTYQGuMbpafMqftalz'
-    RAINFOCUS_WIDGET_ID = 'n6l4Lo05R8fiy3RpUBm447dZN8uNWoye'
-    BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/5647924234001/SyK2FdqjM_default/index.html?videoId=%s'
+    _BASE_URL = 'https://www.ciscolive.com'
+    _TOKEN = None
 
-    HEADERS = {
-        'Origin': 'https://ciscolive.cisco.com',
-        'rfApiProfileId': RAINFOCUS_API_PROFILE_ID,
-        'rfWidgetId': RAINFOCUS_WIDGET_ID,
-    }
-
-    def _call_api(self, ep, rf_id, query, referrer, note=None):
-        headers = self.HEADERS.copy()
-        headers['Referer'] = referrer
-        return self._download_json(
-            self.RAINFOCUS_API_URL % ep, rf_id, note=note,
-            data=urlencode_postdata(query), headers=headers)
-
-    def _parse_rf_item(self, rf_item):
-        event_name = rf_item.get('eventName')
-        title = rf_item['title']
-        description = clean_html(rf_item.get('abstract'))
-        presenter_name = try_get(rf_item, lambda x: x['participants'][0]['fullName'])
-        bc_id = rf_item['videos'][0]['url']
-        bc_url = self.BRIGHTCOVE_URL_TEMPLATE % bc_id
-        duration = float_or_none(try_get(rf_item, lambda x: x['times'][0]['length']))
-        location = try_get(rf_item, lambda x: x['times'][0]['room'])
-
-        if duration:
-            duration = duration * 60
+    def _real_initialize(self):
+        self._TOKEN = try_call(lambda: self._get_cookies(self._BASE_URL)['rfjwt'].value)
+        if not self._TOKEN:
+            self.raise_login_required()
 
-        return {
-            '_type': 'url_transparent',
-            'url': bc_url,
-            'ie_key': 'BrightcoveNew',
-            'title': title,
-            'description': description,
-            'duration': duration,
-            'creator': presenter_name,
-            'location': location,
-            'series': event_name,
-        }
+    def _call_api(self, endpoint, some_id, payload=None, doseq=False, **kwargs):
+        if endpoint != 'search':
+            payload = {'id': some_id}
```

Comment:
Personally I would have gone with `item_id` rather than `some_id` but a matter of taste really.

* should explicitly supplied `payload` override this?
* if `endpoint == 'search', must `payload` be supplied (or `urlencode_postdata()` crashes, but maybe that's an adequate parameter check)?